### ISSUE TITLE
Bundle product with hundreds of options takes minutes to save and timeout (with inventory enabled)

### ISSUE THREAD

**Issue by santerref** [magento/magento2#34732]
### Preconditions (*)
1. Magento version 2.4.3
2. Inventory enabled
3. Bundle product with a lot of options (200 to 500)

### Steps to reproduce (*)
1. Enable the inventory
2. Create a bundle product with 200+ options (children)

### Expected result (*)
1. Product save quickly as usual
2. The indexes are updated for 201 products

### Actual result (*)
1. The plugin `ReindexSourceItemsAfterSaveBundleSelectionPlugin` of the module `module-inventory-bundle-product-indexer` is called after each saveChild of `\Magento\Bundle\Model\LinkManagement` and reindex all the options and the main product. That means for a bundle product with 200 options, the 200 options get reindexed 200 times ! 200 reindex requests of 201 products.
2. The save takes more than 3 to 5 minutes and the server timeout. We can't save our bundle products.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
Verified the issue on Magento 2.4-develop branch and the issue is reproducible.
Unable to create and save bundle product with 200 child options

**Comment by engcom-Alfa** [comment]
Hi @santerref Thanks for your contribution!
The issue is being verified with more clarity. Thanks for your patience!

Hi @engcom-November 
We have re-check what is the re-indexing mode was set while reproducing the issue.
We have to re-confirm based on that as per our last community triage discussion.
Thanks in advance!

**Comment by engcom-November** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
This issue is reproducible when index mode is set to both "Update by schedule" and "Update on Save"

**Comment by engcom-Hotel** [comment]
Hello @santerref,

This doesn’t seem like an issue with the Magento, after debugging this issue we found that when we are adding 200 plus options Magento creates inputs for each option and these inputs depend on the number of options and their children. In our case for each option, Magento creates 50 plus inputs. So total inputs in our case when we are creating 200 plus options would be ( 200 * 50 = 10000 ) which exceeds the [max_input_var](https://www.php.net/manual/en/info.configuration.php) limit and breaks the execution.

To get this work we just updated the PHP max_input_var as per our requirement and we are able to save the bundle product with 200+ options.

Hence, no development is required. We are closing this issue.

Thanks
